geode-core/src/integrationTest/java/org/apache/geode/cache/query/internal/DefaultQueryServiceWithSecurityIntegrationTest.java

```diff
@@ -0,0 +1,193 @@
+
+  @Rule
+  public TestName testName = new TestName();
+
+  @Rule
+  public ServerStarterRule server = new ServerStarterRule()
+      .withProperty(SECURITY_MANAGER, SimpleSecurityManager.class.getName())
+      .withProperty("security-username", "cluster")
+      .withProperty("security-password", "cluster")
+      .withAutoStart();
+
+  private void createAndPopulateRegion(String name, RegionShortcut shortcut, int entries) {
+    Cache cache = server.getCache();
+    Region<Integer, QueryTestObject> region =
+        cache.<Integer, QueryTestObject>createRegionFactory(shortcut).create(name);
+    IntStream.range(0, entries)
+        .forEach(id -> region.put(id, new CustomQueryTestObject(id, "id_" + id)));
+    await().untilAsserted(() -> assertThat(region.size()).isEqualTo(entries));
+  }
+
+  @Before
+  public void setUp() {
+    spiedCache = spy(server.getCache());
+    InternalQueryService spyQueryService = spy(new DefaultQueryService(server.getCache()));
+    doReturn(new SpyAuthorizer()).when(spyQueryService).getMethodInvocationAuthorizer();
+    when(spiedCache.getQueryService()).thenReturn(spyQueryService);
+  }
+
+  @After
+  public void tearDown() {
+    SpyAuthorizer.authorizations.set(0);
+  }
+
+  private void executeQueryAndAssertThatCachedAuthorizationsWereUsed(String queryString,
+      int methodInvocations, int executions, int entries) {
+    // Every new query has a new context and, thus, authorization should be executed again.
+    IntStream.range(1, executions).forEach(counter -> {
+      try {
+        DefaultQuery query = spy(new DefaultQuery(queryString, spiedCache, false));
+
+        // Make sure the PR Executor uses our spy cache.
+        doAnswer(answer -> {
+          QueryExecutor executor = (QueryExecutor) answer.callRealMethod();
+          if (executor instanceof PartitionedRegion) {
+            PartitionedRegion prExecutor = spy((PartitionedRegion) executor);
+            when(prExecutor.getCache()).thenReturn(spiedCache);
+            return prExecutor;
+          }
+
+          return executor;
+        }).when(query).checkQueryOnPR(any());
+
+        SelectResults result = (SelectResults) query.execute();
+        assertThat(result.size()).isEqualTo(entries);
+
+        assertThat(SpyAuthorizer.authorizations.get()).isEqualTo(methodInvocations * counter);
+      } catch (Exception exception) {
+        throw new RuntimeException(exception);
+      }
+    });
+  }
+
+  @Test
+  @Parameters({"LOCAL", "REPLICATE", "PARTITION"})
+  @TestCaseName("[{index}] {method}(RegionType:{0}")
+  public void queryWithSingleProjectionFieldShouldUseCachingForImplicitMethodAuthorizations(
+      RegionShortcut regionShortcut) {
+    String regionName = testName.getMethodName();
+    createAndPopulateRegion(regionName, regionShortcut, 100);
+    String queryString = "SELECT object.name FROM /" + regionName + " object";
+
+    executeQueryAndAssertThatCachedAuthorizationsWereUsed(queryString, 1, 10, 100);
```

Comment:
Consider extracting the entries to a value to ensure consistency between the `createAndPopulateRegion` and `executeQueryAndAssertThatCachedAuthorizationsWereUsed` methods.

Reply:
👍